### PR TITLE
Reinstate tox -e lint

### PR DESCRIPTION
**Pull request by xrmx**
# Description

Reintroduce the lint environment after 5116305f77bcd4c8ab18ef302a4351bb5b724c1e to just run the cheap linting tools so that developers can rely on it

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] tox -e lint

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by ocelotl** [comment]
Do we really need this? :thinking: We already have linting for every package, is there a place where we are not providing linting right now?

File: tox.ini [hunk 1]
```diff
@@ -1189,6 +1189,17 @@ changedir = docs
...
+[testenv:lint]
+basepython: python3
+recreate = True
+deps =
+  -r dev-requirements.txt
+
+commands =
+  black --config {toxinidir}/pyproject.toml {{toxinidir}} --diff --check
+  isort --settings-path {toxinidir}/.isort.cfg {{toxinidir}} --diff --check-only
+  flake8 --config {toxinidir}/.flake8 {toxinidir}
```
Comment (ocelotl) [inline tox.ini:1201]: It's also a bit confusing to have a `lint` test environment that does not run `pylint` like other test environments do...

**Comment by xrmx** [comment]
> Do we really need this? 🤔 We already have linting for every package, is there a place where we are not providing linting right now?

The only reason for this in the CONTRIBUTING hunk, we have to make it simple for people to run this tools otherwise they don't.

File: tox.ini (hunk 1 above)
Comment (xrmx) [reply tox.ini:1201]: If it wasn't a pain to run pylint for a multi package repository I would gladly add it :sweat_smile:

**Review by ocelotl — APPROVED** [review]

**Review by lzchen — APPROVED** [review]

**Pull request merged**